snap/snapcraft.yaml

```diff
@@ -32,6 +37,10 @@ apps:
     daemon: simple
   keytool:
     command: usr/lib/jvm/java-11-openjdk-amd64/bin/keytool
+  transaction-log-toolkit:
+    command: opt/zookeeper/bin/zkTxnLogToolkit.sh
+  snapshot-toolkit:
+    command: opt/zookeeper/bin/zkSnapShotToolkit.sh
```

Comment:
**suggestion:** Were you able to test these against a running charm? I ask, because snap confinement can sometimes make it difficult for snap services to 'read' mounted storage volumes.

If you want to validate this and haven't, you can probably do something like the following:
0. Build the snap locally
1. Locally build the ZooKeeper charm, but with the following changes to `src/charm.py`:
```python
    def _on_install(self, event: InstallEvent) -> None:
        """Handler for the `on_install` event."""
    
        # warning for manual change
        logger.info("Ready to install, copy the built snap over.")
        time.sleep(100)

		# commenting out default snap install
        # install = self.workload.install()
        # if not install:
        #     self._set_status(Status.SERVICE_NOT_INSTALLED)
        #     event.defer()
        #     return
```
2. Deploy the local charm, and when that log appears in `juju debug-log`, copy the locally built snap to the charm and install that one, with something like:
```
# '0' here is the machine id, grabbed from `juju status`
juju scp ./charmed-zookeeper_3.6.4_amd64.snap 0:\
juju ssh zookeeper/0 sudo -i 'sudo snap install /home/ubuntu/charmed-zookeeper_3.6.4_amd64.snap --dangerous
```
3. Then let the sleep play out, and the charm should start on it's own. After that, it will be using the mounted storage, and you can manually run `charmed-zookeeper.transaction-log-toolkit|snapshot-toolkit` to see what you get

You may run in to permission issues, but hopefully that's a good validation in a charm environment if not done already 👍🏾

Reply:
Thanks for the great advice, sure will do that and report the result

Reply:
Tested in zookeeper charm and seems everything works as expected:

![Screenshot from 2024-11-13 12-53-38](https://github.com/user-attachments/assets/c2caea24-1ffc-4d2a-a279-0e6dcf292282)

Reply:
Great! all good then :)